### ISSUE TITLE
[FR] Support OpenAI baseurl/proxies in some form

### ISSUE THREAD

**Issue by AsafMah** [AppFlowy-IO/AppFlowy#3855]
### Description

Currently the openai api points are hardcoded into the code (api.openai.com ...).

This prevents usage over proxies where the normal site may be blocked, or usage with compatible apis such as azure's openai or localai.

I suggest adding a baseurl option along with the openai key option, that will default to the openai endpoint but will let users choose.

### Impact

Users that can't use the normal openai endpoint can still benefit from the AI features in appflowy.

**Comment by Stinosko** [comment]
I'm also interested in this feature!

**Comment by leleogere** [comment]
Yes! That would also allow to use API for Mistral AI for example which is compatible with OpenAI's one! However, it would also require to be able to specify the model to use, not only the baseurl (`mistral-tiny`, `mistral-small` or `mistral-medium`). Being able to change the model would also allow OpenAI users to choose the model they want to use. Currently, `gpt-3.5-turbo-instruct` is hardcoded:

https://github.com/AppFlowy-IO/AppFlowy/blob/90516b6adca50013b776ec1987753f877c1366c8/frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/openai/service/openai_client.dart#L100-L107

**Comment by Medve01** [comment]
It's 2025 and we still cannot use any other AI provider than OpenAI. It's sad. Eg. Azure Openai should not be that huge issue to implement as it's merely a configurable base url...

**Comment by Jasmin25** [comment]
I'd recommend to go with something like https://openrouter.ai/ for once off effort to support all vendors.

**Comment by SolarDesalination** [comment]
Will this be supported? I would like to use Deepseek instead of Claude or Opus.
